### PR TITLE
Add Accelerator specific interfaces and there respective implementations

### LINKED ISSUE #1312: GPU MIG Right sizing recommendations by kruize
### Describe the feature 

Kruize reads CPU & memory usage data from the provided data source and comes up with the CPU and Memory right sizing recommendation. In a similar way it would be good to have the GPU MIG partition sizing recommendation for container which utilise GPU's

### Examples or references

Most of the ML workloads need GPU power and advanced GPU's from NVIDIA support MIG (Multi instance GPU's) where a single Physical GPU can be partitioned into multi instances of virtual or logical GPU's which can be configured and shared across multiple containers. Ampere (from A30) and Hopper series GPU's provide this feature.

### Suggest a solution

- Record the GPU related metrics
- Process the metrics along with CPU and Memory metrics
- Provide MIG partition recommendation

### Additional Context

None

### PR DESCRIPTION
**Pull request by bharathappali**
## Description

This PR is a first's part of the set of PR's which will be raised for adding GPU support #1312 

This PR adds Accelerator specific interfaces and there respective implementations

This PR is build on top of #1313, so it needs to be merged after merging #1313 

Fixes #1312 

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Tests will be added in a subsequent PR

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

None

### CODE REVIEW

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorProfile.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.autotune.common.data.system.info.device.accelerator.metadata;
+
+public class AcceleratorProfile {
+    private final String profileName;
+    private final double memoryFraction;
+    private final double smFraction;
+    private final int instancesAvailable;
+
+    public AcceleratorProfile(String profileName, double memoryFraction, double smFraction, int instancesAvailable) {
```
Comment (dinogun) [inline src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorProfile.java:9]: javadoc here

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+import com.autotune.analyzer.utils.AnalyzerConstants;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AcceleratorMetaDataService {
```
Comment (dinogun) [inline src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java:12]: javadoc here

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    private static void initializeAcceleratorProfiles() {
+        List<AcceleratorProfile> commonProfiles = new ArrayList<>();
+        // IMPORTANT: Add it in the ascending order according to GPU Core and Memory Units as we will break the loop upon getting the right one
+        commonProfiles.add(new AcceleratorProfile("1g.10gb", 1.0 / 8, 1.0 / 7, 7));
+        commonProfiles.add(new AcceleratorProfile("1g.20gb", 1.0 / 4, 1.0 / 7, 4));
+        commonProfiles.add(new AcceleratorProfile("2g.20gb", 2.0 / 8, 2.0 / 7, 3));
+        commonProfiles.add(new AcceleratorProfile("3g.40gb", 4.0 / 8, 3.0 / 7, 2));
+        commonProfiles.add(new AcceleratorProfile("4g.40gb", 4.0 / 8, 4.0 / 7, 1));
+        commonProfiles.add(new AcceleratorProfile("7g.80gb", 1.0, 1.0, 1));
```
Comment (dinogun) [inline src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java:29]: Create constants for all of this. Same for the ones below

File: src/main/java/com/autotune/common/data/system/info/device/ContainerDeviceList.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+
+    @Override
+    public void updateDevice(AnalyzerConstants.DeviceType deviceType, DeviceDetails deviceInfo) {
+        // TODO: Need to be implemented if we need a dynamic experiment device updates
+    }
+
+    @Override
+    public DeviceDetails getDeviceByParameter(AnalyzerConstants.DeviceType deviceType, String matchIdentifier, AnalyzerConstants.DeviceParameters deviceParameters) {
```
Comment (dinogun) [inline src/main/java/com/autotune/common/data/system/info/device/ContainerDeviceList.java:66]: javadoc here for the usecases and document the valid return values

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorProfile.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorProfile.java:19]: @dinogun Thanks for the review. I have added the java doc

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java:22]: Done.

File: src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java (hunk 3 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/accelerator/metadata/AcceleratorMetaDataService.java]: Created the constants for the profile names, excluded the constants for fractions. Should I create constants for fractions as well?

File: src/main/java/com/autotune/common/data/system/info/device/ContainerDeviceList.java (hunk 4 above)
Comment (bharathappali) [reply src/main/java/com/autotune/common/data/system/info/device/ContainerDeviceList.java:80]: Added the javadoc and use case comments

**Comment by dinogun** [comment]
@bharathappali can you please rebase this to mvp_demo instead of master

**Comment by bharathappali** [comment]
Apologies for the overlook, I have rebased to mvp_demo. Thanks @dinogun

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**